src/httpserver/routing/router.clj

```diff
@@ -0,0 +1,15 @@
+(ns httpserver.routing.router)
+
+(defn check-route [method path route]
+  (and (= (:method route) method) (= (:path route) path)))
+
+(defn find-route [method path routes]
+  (some #(when (check-route method path %) %) routes))
+
+(defn get-handler [request routes]
+  (:handler (find-route (:method request) (:path request) routes))
```

Comment:
This reads a little long for a one-liner. On the other hand, a `let` form seems a little heavy-handed...

Alright, it's time to learn about destructuring!

If you know the shape of an argument to your function, you can destructure it into its component parts. See if you can figure out what's going on in the example below—_don't implement it in your code until you understand it!_ There are lots of tutorials/blog posts/examples on the internet about destructuring, and I'm sure some apprentices would be willing to help explain it—it's a fun concept :)

```clojure
(defn get-handler [{:keys [method path]} routes]
  (:handler (find-route method path routes)))
```